modules/maven3/manifests/init.pp

```diff
@@ -5,37 +5,44 @@
 # @example
 #  include maven3
 #
-#  class { 'maven3': 
+#  class { 'maven3':
 #    version => '3.5.3',
 #  }
 
 class maven3 (
   $version = '3.5.3',
+  $load_dir = "/tmp/",
+  $source = "http://archive.apache.org/dist/maven/maven-3/${version}/binaries/",
+  $archive = "apache-maven-${version}-bin.tar.gz",
+  $install_path = "/opt/",
   ){
 
-  $archive = "apache-maven-${version}-bin.tar.gz"
-  $source = "http://archive.apache.org/dist/maven/maven-3/${version}/binaries/apache-maven-${version}-bin.tar.gz"
-  $load_dir = "/tmp/"
   $archive_path = "${load_dir}${archive}"
 
-  exec { 'upload_maven_archive':
-    command => "wget ${source}",
-    path    => '/bin:/sbin:/usr/bin:/usr/sbin:',
-    cwd     => "${load_dir}",
-    creates => "${archive_path}",
-    before  => Exec['maven-untar'],
+  if !("jdk" in $facts['path']) {
+    warning('No JDK installed')
```

Comment:
I have a deja vu.

That check is useless i think.

Reply:
accepted